WordPress/gutenberg
# TextHighlight: Convert component to TypeScript

## What?
Converts the `TextHighlight` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `TextHighlight` to TypeScript.
- Refactors the test to use `@testing-library/react'`
- Refactors the storybook story

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `TextHighlight` continues to function as expected
3. Test the various TextHighlight within the block/site editors in Gutenberg still work
